### PR TITLE
[IR] Implement `convert_attributes`

### PR DESCRIPTION
**Pull request by justinchuby**
Implement `convert_attribute` and `convert_attributes` as convenience functions for the IR.

Also updates repr for Attr to make it more succinct.

### CODE REVIEW

File: onnxscript/ir/_convenience.py [hunk 1]
```diff
@@ -10,30 +10,168 @@
...
+            raise ValueError(
+                f"Attribute type '{attr.type}' does not match provided type '{attr_type}'"
+            )
+        return attr
+
+    if attr_type is None:
+        attr_type = _infer_attribute_type(attr)
+
+    else:
+        if isinstance(attr, (_core.Attr, _core.RefAttr)):
```
Comment (gramalingam) [inline onnxscript/ir/_convenience.py:105]: Checks seem duplicated above and here

**Review by titaiwangms — COMMENTED** [review]
I think it's been used by `generic_pattern.py`, so the file should also be updated?

File: onnxscript/ir/_convenience.py [hunk 2]
```diff
@@ -10,30 +10,168 @@
...
+        return _core.AttrStrings(name, attr)  # type: ignore
+    if attr_type == _enums.AttributeType.TENSOR:
+        if isinstance(attr, (_core.TensorBase, _protocols.TensorProtocol)):
+            return _core.AttrTensor(name, attr)
+        if isinstance(attr, onnx.TensorProto):
+            return _core.AttrTensor(name, serde.TensorProtoTensor(attr))
+    raise TypeError(f"Unsupported attribute type: '{type(attr)}'")
+
+
+def convert_attributes(
```
Comment (gramalingam) [inline onnxscript/ir/_convenience.py:135]: nit: I wonder if we should (later) simplify this double-conversion (from a dictionary/map to list here, and then back from list to dictionary when we construct a node)?

**Review by gramalingam — APPROVED** [review]

File: onnxscript/ir/_convenience.py (hunk 1 above)
Comment (justinchuby) [reply onnxscript/ir/_convenience.py:105]: Good catch thanks

**Comment by justinchuby** [comment]
> I think it's been used by `generic_pattern.py`, so the file should also be updated?

I think it's updated? In _tape right?

**Pull request merged**